ibis/formats/pandas.py

```diff
@@ -247,38 +250,78 @@ def convert(values, names=dtype.names, converters=converters):
 
         return convert
 
-    @staticmethod
-    def convert_JSON_element(_):
-        def try_json(x):
-            if x is None:
-                return x
+    @classmethod
+    def convert_JSON_element(cls, _):
+        import json
+
+        def convert(value):
+            if value is None:
+                return value
             try:
-                return json.loads(x)
+                return json.loads(value)
             except (TypeError, json.JSONDecodeError):
-                return x
-
-        return try_json
-
-    @staticmethod
-    def convert_Array_element(dtype):
-        convert_value = PandasData.get_element_converter(dtype.value_type)
-        return lambda values: [
-            convert_value(value) if value is not None else value for value in values
-        ]
-
-    @staticmethod
-    def convert_Map_element(dtype):
-        convert_value = PandasData.get_element_converter(dtype.value_type)
-        return lambda row: {
-            key: convert_value(value) if value is not None else value
-            for key, value in dict(row).items()
-        }
-
-    @staticmethod
-    def convert_UUID_element(_):
+                return value
+
+        return convert
+
+    @classmethod
+    def convert_Timestamp_element(cls, dtype):
+        def converter(value, dtype=dtype):
+            if value is None:
+                return value
+
+            with contextlib.suppress(AttributeError):
+                value = value.item()
+
+            if (tz := dtype.timezone) is not None:
+                return value.astimezone(normalize_timezone(tz))
+
+            return value.replace(tzinfo=None)
+
+        return converter
+
+    @classmethod
+    def convert_Array_element(cls, dtype):
+        convert_value = cls.get_element_converter(dtype.value_type)
+
+        def convert(values):
+            if values is None:
+                return values
+
+            return [
+                convert_value(value) if value is not None else value for value in values
+            ]
+
+        return convert
+
+    @classmethod
+    def convert_Map_element(cls, dtype):
```

Comment:
Just mentioning that the logic here for object dtypes is basically identical to what we have in `dt.normalize()` so I would prefer to reduce the code duplication in the future.